### PR TITLE
B 21536 int

### PR DESCRIPTION
**Pull request by CoryKleinjanCACI**
## Summary

When a TOO approves multiple requested update to a shipments address, created by Prime, the Move History tab was only displaying the most recent approval. This was due to some columns being ignored, and the address being replaced by ID rather than being actually updated. 

Another issue, found by @MInthavongsay, where Secondary addresses boolean being set to 'No' after having previously been 'Yes' was delisting old logs related to secondary addresses was fixed as well. The test steps have been updated to reflect these changes.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Setup: Create a shipment with an HHG and push it through the process until it is available to prime.

1. Access the Prime Sim
2. Access your create HHG
3. Request to Update Shipment Destination Address (Be sure the new address is in a different service area so it requires TOO approval)
4. Login as a TOO
5. Navigate to the created HHG
6. Select to edit the shipment
7. Select to review the shipment destination address change
8. Approve the shipment destination address change
9. Navigate to the move history tab and validate that the new address update shows on the page.
10. Repeat steps 1-9 again and validate both updates are still present in move history

Test steps for secondary addresses
1. Login as a TOO
2. Navigate to a move with an HHG and add a secondary address.
3. Save the shipment.
4. View the Move History and see that it displayed the added shipment
5. Remove the secondary address from the shipment
6. View the Move History and see that it displays both the addition and removal of the secondary address
7. Repeat steps 2-6 multiple times on the same shipment to verify logs are retained

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
Screenshot 2024-12-10 at 11 00 41 AM

### CODE REVIEW

File: migrations/app/schema/20221129184114_redifine_audit_tables.up.sql [hunk 1]
```diff
@@ -83,10 +83,6 @@ SELECT add_audit_history_table(
 	ignored_cols := ARRAY[
 		'created_at',
 		'updated_at',
-		'destination_address_id',
```
Comment (MInthavongsay) [inline migrations/app/schema/20221129184114_redifine_audit_tables.up.sql:86]: For this particular change can this be done in a new migration file? Or updating the original migration the only way?

**Review by deandreJones — CHANGES_REQUESTED** [review]
new migration file needed

**Review by MInthavongsay — CHANGES_REQUESTED** [review]
Need to create a new migration file to apply the db changes instead of updating existing migration file.

File: migrations/app/schema/20221129184114_redifine_audit_tables.up.sql (hunk 1 above)
Comment (CoryKleinjanCACI) [reply migrations/app/schema/20221129184114_redifine_audit_tables.up.sql:86]: Nope. Just forgot. Fixing it now.

**Review by MInthavongsay — APPROVED** [review]
Talked to Cory about some "issues" uncovered that are related to audit logs that are outside the scope of this BL. For example,  DELETEs of addresses are not correctly captured in the audit logs. This will need to be addressed in another BL and/or EPIC.  Will approve this per AC.

**Review by antgmann — APPROVED** [review]
New migration file looks good, testing steps work as instructed.

File: src/constants/MoveHistory/Database/FieldMappings.js [hunk 2]
```diff
@@ -44,10 +44,14 @@ export default {
...
   pickup_address: 'Pickup Address',
   secondary_pickup_address: 'Second Pickup Address',
+  has_secondary_pickup_address: 'Secondary Pickup Address',
   tertiary_pickup_address: 'Third Pickup Address',
+  has_tertiary_pickup_address: 'Third Pickup Address',
   destination_address: 'Delivery Address',
   secondary_destination_address: 'Second Delivery Address',
+  has_secondary_delivery_address: 'Secondary Delivery Address',
   tertiary_destination_address: 'Third Delivery Address',
+  has_tertiary_destination_address: 'Third Delivery Address',
```
Comment (danieljordan-caci) [inline src/constants/MoveHistory/Database/FieldMappings.js:54]: `delivery` or `destination`? If they're different then I hate it.

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Discrepancy i n variable names

File: pkg/assets/sql_scripts/move_history_fetcher.sql [hunk 3]
```diff
@@ -1,7 +1,7 @@
 WITH move AS (
 		SELECT
 			moves.*
-		FROM
+				FROM
```
Comment (danieljordan-caci) [inline pkg/assets/sql_scripts/move_history_fetcher.sql:4]: Can you put this back to where it was?

Comment (CoryKleinjanCACI) [reply pkg/assets/sql_scripts/move_history_fetcher.sql]: Yea, I think pre-commit might have done that for some reason. Just tried pushing a change to remove the weird spacing and it took it now.

File: src/constants/MoveHistory/Database/FieldMappings.js (hunk 2 above)
Comment (CoryKleinjanCACI) [reply src/constants/MoveHistory/Database/FieldMappings.js:54]: This is how it exists in existing logs, and other places in the application. The backend is secondary_destination, but the frontend displays delivery. It is like this on the Approved Shipments tab of a move as well. I also don't like it.

Comment (danieljordan-caci) [reply src/constants/MoveHistory/Database/FieldMappings.js:54]: Gross.

**Review by danieljordan-caci — APPROVED** [review]

File: src/constants/MoveHistory/Database/FieldMappings.js (hunk 2 above)
Comment (CoryKleinjanCACI) [reply src/constants/MoveHistory/Database/FieldMappings.js:54]: Screenshot 2024-12-12 at 3 09 35 PM

**Comment by antgmann** [comment]
Accepted happo diff

**Review by antgmann — APPROVED** [review]

**Review by deandreJones — APPROVED** [review]

**Pull request merged**